File: 104 Maximum Depth of Binary Tree/Maximum Depth of Binary Tree.java
// Recursive DFS (Depth-First Search)
class Solution {
    public int maxDepth(TreeNode root) {
        if (root == null) {
            return 0;
        }
        return Math.max(maxDepth(root.left), maxDepth(root.right)) + 1;
    }

}

    // BFS (Breadth-First Search)
public int maxDepth(TreeNode root) {
    if (root == null) {
        return 0;
    }
    Queue<TreeNode> q = new LinkedList<>();
    q.offer(root);
    int level = 0;
    
    while (!q.isEmpty()) {
        int count = q.size();
        for (int i = 0; i < count; i++) {
            TreeNode node = q.poll();
            
            if (node.left != null) {
                q.offer(node.left);
            }
            if (node.right != null) {
                q.offer(node.right);
            }
        }
        level++;
    }
    return level;
}

File: 104 Maximum Depth of Binary Tree/Maximum Depth of Binary Tree.js
//Recursive DFS (Depth-First Search)
var maxDepth = function (root) {
  if (root === undefined || root === null) {
    return 0;
  }
  return Math.max(maxDepth(root.left), maxDepth(root.right)) + 1;
};

//BFS (Breadth-First Search)
var maxDepth = function (root) {
    if (root === null) {
        return 0;
    }
    var q = [root];
    var level = 0;
    
    while (q.length != 0) {
        var count = q.length;
        for (a = 0; a < count ; a++) {
            var node = q.shift();
                        
            if (node.left != null) {
                q.push(node.left);
            }
            if (node.right != null) {
                q.push(node.right);
            }
        }
        level++;
    }
    return level;
};
